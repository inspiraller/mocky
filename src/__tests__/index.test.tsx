import { Component } from 'react';
import Index, { render } from 'src/index';

class Comp extends Component {
  componentDidMount() {
    document.title = 'Hello vanilla webpack 4 and all typical packages in a project';
  }

  render() {
    return null;
  }
}

const DOM = {
  render: jest.fn()
};
const originalGetElement = global.document.getElementById;

describe('index.tsx', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  describe('initialisation', () => {
    it('renders without crashing', () => {
      const obj = { ...Index, _reactInternalInstance: 'censored' };
      const strObj = JSON.stringify(obj);
      expect(strObj).toMatchSnapshot();
    });
  });

  describe('render', () => {
    beforeEach(() => {
      global.document.getElementById = () => null;
    });
    beforeEach(() => {
      global.document.getElementById = originalGetElement;
    });
    it('should call render', () => {
      render({ DOM, Comp });
      expect(DOM.render).toHaveBeenCalledTimes(1);
    });
  });
});
