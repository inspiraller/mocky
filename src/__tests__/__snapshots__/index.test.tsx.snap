// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.tsx initialisation renders without crashing 1`] = `"{\\"_reactInternalInstance\\":\\"censored\\"}"`;
